packages/pigeon/lib/java_generator.dart

```diff
@@ -514,9 +533,11 @@ class JavaGenerator extends StructuredGenerator<JavaOptions> {
   ///   int add(int x, int y);
   void _writeInterfaceMethod(JavaOptions generatorOptions, Root root,
       Indent indent, Api api, final Method method) {
+    final String nullableType =
+        method.isAsynchronous ? '' : _nullSafeString(method.returnType);
     final String returnType = method.isAsynchronous
         ? 'void'
-        : _nullsafeJavaTypeForDartType(method.returnType);
+        : _javaTypeForDartType(method.returnType);
```

Comment:
Isn't this a functional change?

Reply:
nope, the "change" is accounted for in the later code